### PR TITLE
Fix Multi Select Crash

### PR DESCRIPTION
**Pull request by alperozturk96**
- [x] Tests written, or not not needed

### CODE REVIEW

File: app/src/main/java/com/owncloud/android/ui/adapter/OCFileListAdapter.java [hunk 1]
```diff
@@ -289,21 +293,61 @@ public void setEncryptionAttributeForItemID(String fileId, boolean encrypted) {
...
+
+        if (isFilesEmpty) {
+            if (shouldShowHeader()) {
+                return headerId;
+            } else {
+                return footerId;
+            }
+        } else {
+            return mFiles.get(position).getFileId();
         }
```
Comment (ZetaTom) [inline app/src/main/java/com/owncloud/android/ui/adapter/OCFileListAdapter.java:308]:
```suggestion
        if (shouldShowHeader()) {
            if (position == 0) {
                return headerId;
            }

            // skip header
            position--;
        }

        if (position == mFiles.size()) {
            return footerId;
        } if (position < mFiles.size()) {
            return mFiles.get(position).getFileId();
        }

        // fallback
        return RecyclerView.NO_ID;
```

With this patch in place, it no longer crashes on my test device.

**Review by tobiasKaminsky — APPROVED** [review]

**Review by JonasMayerDev — APPROVED** [review]
Works for me

**Comment by joshtrichards** [comment]
Was this for #11797?

**Comment by ZetaTom** [comment]
> Was this for #11797?

No, this PR seved to fix a more generic issue, that would occur when selecting the last item in a given directory. Previously, I've managed to reproduce the issue using the following steps:

1. set up new Nextcloud user in app using default skeleton
2. open *All files* tab in the app
3. long press on the second to last item in the root directory (e.g. *Reasons to use Nextcloud.pdf*)
4. add the last item (e.g. *Template credits.md*) to the selection

Following these steps would crash the app with the following exception:

```
java.lang.IllegalStateException: Two different ViewHolders have the same stable ID. Stable IDs in your adapter MUST BE unique and SHOULD NOT change.
```

The exception was caused by `OCFileListAdapter:getItemId()` returning incorrect ids in some cases. These were in conflict with `setHasStableIds(true);`, which requires each item in the recycler list view to have a unique and persistent id.

**Pull request merged**